dashboard/app/models/foorm/submission.rb

```diff
@@ -11,5 +11,82 @@
 #
 
 class Foorm::Submission < ActiveRecord::Base
-  belongs_to :foorm_form
+  has_one :metadata, class_name: 'Pd::WorkshopSurveyFoormSubmission', foreign_key: :foorm_submission_id
+  belongs_to :form, foreign_key: [:form_name, :form_version], primary_key: [:name, :version]
+
+  SURVEY_CONFIG_KEYS = %w(
```

Comment:
Not sure this is exactly the right list...

Reply:
for the regular workshop surveys you can reference this method: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/pd/workshop_daily_survey_controller.rb#L558, only one that's wrong is it should be `workshop_agenda` instead of `agenda`

Reply:
Should I actually call that method and look at the keys of the returned hash? Or do you mean just create a new constant that matches the list in that controller?

Reply:
I'm not sure what the best approach here is. This will all break when we get to misc surveys because we allow any variables to be sent via url parameters for those. I think eventually we will need to pull the variables from the survey submission itself.

Reply:
Ah ok, makes sense -- I guess I can just say `unless general_parsed_questions.keys.include? question_id` instead of using the `SURVEY_CONFIG_KEYS` list, which should catch anything like that?

Reply:
I think so!